vllm/distributed/kv_transfer/kv_connector/utils.py

```diff
@@ -94,9 +93,11 @@ def put_kv_to_cache(self, model_executable: torch.nn.Module, keys, values,
 
 
 def get_kv_connector_cache_layout():
+    # NOTE (NickLucche) When running disaggregated PD with NIXL, HND layout is
+    # used for faster transfer.
     vllm_config = get_current_vllm_config()
     kv_config = vllm_config.kv_transfer_config
-    if vllm_config.model_config is None:
+    if vllm_config.model_config is None or kv_config is None:
```

Comment:
If the `logger.warning` and `logger.info` are very spammy and printing for every layer, we could change them to `logger.warning_once` and `logger.info_once`